chapters/sparse.rst

```diff
@@ -0,0 +1,287 @@
+How to use Sparse Embeddings in Jina
+=====================================
+
+.. meta::
+   :description: How to use Sparse Embeddings in Jina
+   :keywords: Jina, sparse, coo, csr
+
+.. contents:: Table of Contents
+    :depth: 2
+
+Motivation
+------------
+
+A sparse matrix is a special case of a matrix in which the number of zero elements is much higher than the number of non-zero elements.
+The space used for representing data and the time for scanning the matrix can be reduced significantly using a sparse representation.
+In this guideline, we'll introduce how to use sparse matrix in Jina.
+
+Before you start
+----------------
+
+Before you begin, make sure you meet these prerequisites:
+
+-  Make sure you have basic understanding on Jina.
+-  Make sure you have basic understanding on sparse matrices.
+-  We assume you have some experience with `sparse module in scipy <https://docs.scipy.org/doc/scipy/reference/sparse.html>`_, or you have used `TensorFlow Sparse Tensor <https://www.tensorflow.org/api_docs/python/tf/sparse/SparseTensor>`_ / `PyTorch Sparse COO Tensor <https://pytorch.org/docs/stable/sparse.html#sparse-coo-tensors>`_.
+
+And have Jina installed on your machine:
+
+.. highlight:: shell
+.. code-block:: shell
+
+    pip install "jina[hub]"
+
+
+Behind Jina Sparse Matrix
+-------------------------
+
+As a framework of search, Jina don't have a native sparse matrix support.
+The `Ndarray.sparse` module in Jina's Primitive Types is an adapter between Jina and other sparse backends,
+such as `Scipy.sparse`.
+In Jina, we support three `backends` to create your sparse matrix/Tensor:
+`Scipy`, `TensorFlow` and `Pytorch`.
+You might noticed that `Scipy.sparse` supports different sparse formats,
+while Jina only supports `COO`, `BSR`, `CSR` and `CSC`.
+
+When creating your own sparse matrix,
+we suggest you use `CSR` as default matrix type.
+
+.. list-table:: Sparse Matrix Formats
+   :widths: 25 25 25 25 25
+   :header-rows: 1
+
+   * - ShortName
+     - FullName
+     - Scipy
+     - TensorFlow
+     - Pytorch
+   * - COO
+     - COOrdinate format
+     - Yes
+     - Yes
+     - Yes
+   * - BSR
+     - Block Sparse Row matrix
+     - Yes
+     - No
+     - No
+   * - CSC
+     - Compressed Sparse Column matrix
+     - Yes
+     - No
+     - No
+   * - CSR
+     - Compressed Sparse Row matrix
+     - Yes
+     - No
+     - No
+   * - DIA
+     - Sparse matrix with DIAgonal storage
+     - Yes
+     - No
+     - No
+   * - DOK
+     - Dictionary Of Keys based sparse matrix.
+     - Yes
+     - No
+     - No
+   * - LIL
+     - Row-based list of lists sparse matrix
+     - Yes
+     - No
+     - No
+
+
+Build your Sparse Pipeline In Action
```

Comment:
How to build a sparse pipeline in Jina 
(Header of documents are title case, subheading are sentence case)